extensions/Worker.Extensions.Shared/Reflection/ParameterBinder.cs

```diff
@@ -14,6 +15,12 @@ namespace Microsoft.Azure.Functions.Worker.Extensions
     /// </summary>
     internal static class ParameterBinder
     {
+        /// <summary>
+        /// Read only property that contains all of the generic interfaces implemented by <see cref="System.Collections.Generic.List"/>.
+        /// </summary>
+        /// <remarks>This property calls ToList at the end to force the resolution of the LINQ methods that leverage deferred execution.</remarks>
+        private static readonly IEnumerable<Type> validListInterfaceTypes = typeof(List<>).GetInterfaces().Where(t => t.IsGenericType).Select(t => t.GetGenericTypeDefinition()).ToList();
```

Comment:
for performance, can make this a `HashSet<Type>`:

``` CSharp
new HashSet<Type>(typeof(List<>).GetInterfaces().Where(t => t.IsGenericType).Select(t => t.GetGenericTypeDefinition()));
```

Can then use `.Contains` in the `IsListInterface` check
```

Reply:
I like it and will update my PR with it.

The thing it doesn't solve is that building the HashSet is always going to be a performance hit as the `List<>` type will be reflected on to get the types then the LINQ queries will execute.  I don't really see a way to make that aspect faster beyond replacing the LINQ code or going to a static list.

Reply:
Being static that will happen only once, so not a big concern.